devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,579 @@
+			[
+				completedSetup: true,
+				label         : buttonLabel[buttonNumber - 1],
+				isComponent   : true,
+				componentName : "button$buttonNumber",
+				componentLabel: buttonLabel[buttonNumber - 1]
+			])
+
+		child.sendEvent(name: "supportedButtonValues", value: ["pushed"].encodeAsJSON(), displayed: false)
+		child.sendEvent(name: "numberOfButtons", value: 1, displayed: false)
+		child.sendEvent(name: "button", value: "pushed", data: [buttonNumber: 1], displayed: false)
+	}
+}
+
+def sendButtonEvent(gesture, buttonNumber) {
+	def event = createEvent([name: "button", value: gesture, data: [buttonNumber: buttonNumber], isStateChange: true])
+	String childDni = "${device.deviceNetworkId}:$buttonNumber"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+	return createEvent([name: "button", value: gesture, data: [buttonNumber: buttonNumber], isStateChange: true, displayed: false])
+}
+
+def labelForGesture(gesture, attribute) {
+	if (attribute == 0) {
+		gesture;
+	} else {
+		def number = attribute - 1;
+		"${gesture}_${number}x";
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+	log.info("CentralSceneNotification, keyAttributes=${cmd.keyAttributes}, sceneNumber=${cmd.sceneNumber}")
+	def singleClick = 0;
+	def multipleClicks = [3, 4, 5, 6]
+	def supportedAttributes = [singleClick] + multipleClicks
+	int attribute = cmd.keyAttributes
+	int scene = cmd.sceneNumber
+	if (scene == 1 && attribute in supportedAttributes) {
+		sendButtonEvent(labelForGesture("down", attribute), 2);
+	} else if (scene == 2 && attribute in supportedAttributes) {
+		sendButtonEvent(labelForGesture("up", attribute), 1);
+	} else if (scene == 3 && attribute == singleClick) {
+		sendButtonEvent("pushed", 3)
+	} else {
+		log.warn("Unhandled scene notification, keyAttributes=${attribute}, sceneNumber=${scene}")
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	// Handles all Z-Wave commands we aren't interested in
+	log.debug "${cmd}"
+	[:]
+}
+
+def childSetColor(deviceNetworkId, value) {
+	def id = deviceNetworkId?.split(":")[1] as Integer
+	sendHubCommand setColorCmd(value, id)
+}
+
+def setColorCmd(value, id) {
+	if (value.hue == null || value.saturation == null) return
+	def ledColor = Math.round(huePercentToZwaveValue(value.hue))
+	encapSequence([
+		zwave.configurationV2.configurationSet(scaledConfigurationValue: ledColor, parameterNumber: 13, size: 2),
+		zwave.configurationV2.configurationGet(parameterNumber: 13)
+	], 1000)
+}
+
+private huePercentToZwaveValue(value) {
+	return value <= 2 ? 0 : (value >= 98 ? 255 : value / 100 * 255)
+}
+
+private zwaveValueToHuePercent(value) {
+	return value <= 2 ? 0 : (value >= 254 ? 100 : value / 255 * 100)
+}
+
+def childRefresh(deviceNetworkId) {
+	def id = deviceNetworkId?.split(":")[1] as Integer
```

Comment:
`id` here is unused and also this method doesn't actually refresh a specific child's state.

Reply:
I deleted that, done